<?xml version="1.0" encoding="UTF-8"?> 
<launch>
	<node pkg="kiwi-drive" type="twist_to_motors" name="twist_to_motors" output="screen">
        <param name="motor1" value="/wheel1_vtarget" />
        <param name="motor2" value="/wheel2_vtarget" />
        <param name="motor3" value="/wheel3_vtarget" />
	</node>
	<node pkg="kiwi-drive" type="pid_velocity.py" name="pid_velocity1" output="screen">
        <param name="wheel" value="/enc1" />
        <param name="wheel_vtarget" value="/wheel1_vtarget" />
        <param name="motor_cmd" value="/motor1_cmd" />
        <param name="wheel_vel" value="/wheel1_vel" />
	</node>
	<node pkg="kiwi-drive" type="pid_velocity.py" name="pid_velocity2" output="screen">
        <param name="wheel" value="/enc2" />
        <param name="wheel_vtarget" value="/wheel2_vtarget" />
        <param name="motor_cmd" value="/motor2_cmd" />
        <param name="wheel_vel" value="/wheel2_vel" />
	</node>
	<node pkg="kiwi-drive" type="pid_velocity.py" name="pid_velocity3" output="screen">
        <param name="wheel" value="/enc3" />
        <param name="wheel_vtarget" value="/wheel3_vtarget" />
        <param name="motor_cmd" value="/motor3_cmd" />
        <param name="wheel_vel" value="/wheel3_vel" />
    </node>
	<node pkg="kiwi-drive" type="diff_tf.py" name="diff_tf" output="screen">
        <param name="encoder1" value="/enc1" />
        <param name="encoder2" value="/enc2" />
        <param name="encoder3" value="/enc3" />
    </node>
	<node pkg="trinity_pi" type="getRobotPose.py" name="poseService" output="screen" />
    <!--
	<node pkg="robot_pose_ekf" type="robot_pose_ekf" name="robot_pose_ekf">
		<param name="output_frame" value="odom" />
		<param name="freq" value="2.0" />
		<param name="sensor_timeout" value="1.0" />
		<param name="odom_used" value="true" />
		<param name="imu_used" value="true" />
		<param name="vo_used" value="false" />
		<param name="debug" value="true" />
		<param name="self_diagnose" value="true" />
    </node> -->
</launch>
